### PR TITLE
python312Packages.pypdf: 4.2.0 -> 4.3.1

### PR DESCRIPTION
**Pull request by javaes**
## Description of changes
https://github.com/py-pdf/pypdf/releases/tag/4.3.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Scrumplex** [comment]
Is the comicon update supposed to be here?

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 328940` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.llama-index</li>
    <li>python311Packages.llama-index-readers-file</li>
    <li>python311Packages.llama-index-readers-file.dist</li>
    <li>python311Packages.llama-index-readers-s3</li>
    <li>python311Packages.llama-index-readers-s3.dist</li>
    <li>python311Packages.llama-index.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
  </ul>
</details>
<details>
  <summary>60 packages built:</summary>
  <ul>
    <li>alpaca</li>
    <li>calibre-web</li>
    <li>calibre-web.dist</li>
    <li>cewler</li>
    <li>cewler.dist</li>
    <li>comic-mandown</li>
    <li>comic-mandown.dist</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>diffoscopeMinimal</li>
    <li>diffoscopeMinimal.dist</li>
    <li>diffoscopeMinimal.man</li>
    <li>mdbook-pdf-outline</li>
    <li>mdbook-pdf-outline.dist</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>pretix</li>
    <li>pretix.dist</li>
    <li>python311Packages.camelot</li>
    <li>python311Packages.camelot.dist</li>
    <li>python311Packages.comicon</li>
    <li>python311Packages.comicon.dist</li>
    <li>python311Packages.fpdf2</li>
    <li>python311Packages.fpdf2.dist</li>
    <li>python311Packages.mandown</li>
    <li>python311Packages.mandown.dist</li>
    <li>python311Packages.pypdf</li>
    <li>python311Packages.pypdf.dist</li>
    <li>python311Packages.pypdf.doc</li>
    <li>python311Packages.xhtml2pdf</li>
    <li>python311Packages.xhtml2pdf.dist</li>
    <li>python312Packages.camelot</li>
    <li>python312Packages.camelot.dist</li>
    <li>python312Packages.comicon</li>
    <li>python312Packages.comicon.dist</li>
    <li>python312Packages.fpdf2</li>
    <li>python312Packages.fpdf2.dist</li>
    <li>python312Packages.mandown</li>
    <li>python312Packages.mandown.dist</li>
    <li>python312Packages.pypdf</li>
    <li>python312Packages.pypdf.dist</li>
    <li>python312Packages.pypdf.doc</li>
    <li>tests.dotnet.structured-attrs.check-output</li>
    <li>tests.dotnet.use-dotnet-from-env.fallback</li>
    <li>tests.dotnet.use-dotnet-from-env.use-dotnet-path-env</li>
    <li>tests.dotnet.use-dotnet-from-env.use-dotnet-root-env</li>
    <li>tests.substitute.legacySingleArg</li>
    <li>tests.substitute.legacySingleReplace</li>
    <li>tests.substitute.legacyString</li>
    <li>tests.substitute.legacyVar</li>
    <li>tests.substitute.substitutions</li>
    <li>tests.testers.testEqualContents.emptyFileAndDir</li>
    <li>tests.testers.testEqualContents.equalDir</li>
    <li>tests.testers.testEqualContents.equalExe</li>
    <li>tests.testers.testEqualContents.fileDiff</li>
    <li>tests.testers.testEqualContents.fileMissing</li>
    <li>tests.testers.testEqualContents.nonExistentPath</li>
    <li>tests.testers.testEqualContents.unequalExe</li>
    <li>tests.testers.testEqualContents.unequalExeInDir</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/python-modules/pypdf/default.nix [hunk 1]
```diff
@@ -36,7 +36,7 @@ buildPythonPackage rec {
     rev = "refs/tags/${version}";
     # fetch sample files used in tests
     fetchSubmodules = true;
-    hash = "sha256-ksLpxfRxrNVXezF0VjbAqadpF6bv/SAOOnCKabhugo0=";
+    hash = "sha256-wSF20I5WaxRoN0n0jxB5O3mAAIOxP/TclYBTRAUwYHo=";
   };
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pypdf/default.nix:40]: Wait a second, we should update the substituteInPlace

File: pkgs/development/python-modules/pypdf/default.nix [hunk 2]
```diff
@@ -27,7 +27,7 @@
 
 buildPythonPackage rec {
   pname = "pypdf";
-  version = "4.2.0";
+  version = "4.3.1";
   format = "pyproject";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pypdf/default.nix:31]: Also we can just set pyproject = true here

**Pull request merged**